src/sentry/dynamic_sampling/prioritise_projects.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import time
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import Mapping, Sequence
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Granularity,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry.sentry_metrics.indexer.strings import TRANSACTION_METRICS_NAMES
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+MAX_SECONDS = 60
+CHUNK_SIZE = 1000
+
+
+def fetch_projects_with_total_volumes() -> Mapping[int, Sequence[Sequence[int]]]:
+    """
+    This function fetch with pagination orgs and projects with count per root project
+    """
+    aggregated_projects = defaultdict(list)
+    start_time = time.time()
+    offset = 0
+    while (time.time() - start_time) < MAX_SECONDS:
+        query = (
+            Query(
+                match=Entity(EntityKey.GenericOrgMetricsCounters.value),
+                select=[
+                    Function("sum", [Column("value")], "root_count_value"),
+                    Column("org_id"),
+                    Column("project_id"),
+                ],
+                groupby=[Column("org_id"), Column("project_id")],
+                where=[
+                    Condition(Column("timestamp"), Op.GTE, datetime.utcnow() - timedelta(hours=6)),
+                    Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                    Condition(
+                        Column("metric_id"),
+                        Op.EQ,
+                        TRANSACTION_METRICS_NAMES[TransactionMRI.COUNT_PER_ROOT_PROJECT.value],
```

Comment:
I need to dedicate to somebody else such as @evanh on the query, I cannot give signoff but it looks ok to me.

What I would like to see here is that we extend string indexer interface instead of importing specific constants.

Reply:
This seems fine to me. One thing that would be good would be to instrument this with some timing metrics so you know how long this loop takes.

Reply:
Yeah it's a good idea, I'll add something. We have metrics around celery task per org should be the same. But we can add to query separately.